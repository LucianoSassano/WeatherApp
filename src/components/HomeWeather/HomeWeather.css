*, ::before, ::after {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }

  


  @media all and (min-width: 280px){
    .container {
      display: flex;
      flex-direction: column;
      position: relative;
      text-align: center;
      height: 25vh;
      width: 100vw;
      align-content: center;
      background-color:RGBA(43,45,66,0.65);
      color: #eae6ec;
      
   
    }
    
    .bloc-tabs {
      display: flex;
      width: 41vw;
    }
  
    .tabs {
      color: #eae6ec;
      padding: 2vw;
      text-align: center;
      width: 50%;
      background: rgba(128, 128, 128, 0.210);
      cursor: pointer;
      border-bottom: 1px solid rgba(0, 0, 0, 0.274);
      box-sizing: content-box;
      position: relative;
      outline: none;
    }
  
    .tabs:not(:last-child){
      border-right: 1px solid rgba(0, 0, 0, 0.274);
    }
    
    .active-tabs  {
      background-color:RGBA(43,45,66,0.85);;
      border-bottom: 1px solid transparent;
    }
    
    .active-tabs::before {
      content: "";
      display: block;
      position: absolute;
      top: -5px;
      left: 50%;
      transform: translateX(-50%);
      width: calc(100% + 2px);
      height: 5px;
      background: rgb(88, 147, 241);
    }
    
    

    button {
      border: none;
      
    }
    .content-tabs {
      flex-grow : 1;
    }
    .content {
      background-color:RGBA(43,45,66, 0.85);
      padding: 20px;
      width: 100%;
      height: 100%;
      display: none;
    }
    .content h2 {
      padding: 0px 0 5px 0px;
    }
    .content hr {
      width: 100px;
      height: 2px;
      background: #222;
      margin-bottom: 5px;
    }
    .content p {
      width: 100%;
      height: 100%;
    }
    .active-content {
      display: block;
    }
    
  } 
  
  @media all and (min-width: 320px){

    .container {
      display: flex;
      flex-direction: column;
      position: relative;
      text-align: center;
      height: 25vh;
      width: 100vw;
      align-content: center;
      background-color:RGBA(43,45,66,0.65);
      color: #eae6ec;

    }
    
    .bloc-tabs {
      display: flex;
      width: 90vw;
    }
  
    .tabs {
      color: #eae6ec;
      padding: 2vw;
      text-align: center;
      width: 50%;
      background: rgba(128, 128, 128, 0.210);
      cursor: pointer;
      border-bottom: 1px solid rgba(0, 0, 0, 0.274);
      box-sizing: content-box;
      position: relative;
      outline: none;
    }
  
    .tabs:not(:last-child){
      border-right: 1px solid rgba(0, 0, 0, 0.274);
    }
    
    .active-tabs  {
      background-color:RGBA(43,45,66,0.85);;
      border-bottom: 1px solid transparent;
    }
    
    .active-tabs::before {
      content: "";
      display: block;
      position: absolute;
      top: -5px;
      left: 50%;
      transform: translateX(-50%);
      width: calc(100% + 2px);
      height: 5px;
      background: rgb(88, 147, 241);
    }
    
    

    button {
      border: none;
      
    }
    .content-tabs {
      flex-grow : 1;
    }
    .content {
      background-color:RGBA(43,45,66, 0.85);
      padding: 20px;
      width: 100%;
      height: 100%;
      display: none;
    }
    .content h2 {
      padding: 0px 0 5px 0px;
    }
    .content hr {
      width: 100px;
      height: 2px;
      background: #222;
      margin-bottom: 5px;
    }
    .content p {
      width: 100%;
      height: 100%;
    }
    .active-content {
      display: block;
    }

  } 
  
  @media all and (min-width: 375px){

    .homeWeather-card{
      width: 80vw;
    }

    .container {
      display: flex;
      flex-direction: column;
      position: relative;
      text-align: center;
      height: 25vh;
      width: 100vw;
      align-content: center;
      background-color:RGBA(43,45,66,0.65);
      color: #eae6ec;
      
      
    }
    
    .bloc-tabs {
      display: flex;
      width: 80vw;
    }
  
    .tabs {
      color: #eae6ec;
      padding: 2vw;
      text-align: center;
      width: 50%;
      background: rgba(128, 128, 128, 0.210);
      cursor: pointer;
      border-bottom: 1px solid rgba(0, 0, 0, 0.274);
      box-sizing: content-box;
      position: relative;
      outline: none;
    }
  
    .tabs:not(:last-child){
      border-right: 1px solid rgba(0, 0, 0, 0.274);
    }
    
    .active-tabs  {
      background-color:RGBA(43,45,66,0.85);;
      border-bottom: 1px solid transparent;
    }
    
    .active-tabs::before {
      content: "";
      display: block;
      position: absolute;
      top: -5px;
      left: 50%;
      transform: translateX(-50%);
      width: calc(100% + 2px);
      height: 5px;
      background: rgb(88, 147, 241);
    }
    
    

    button {
      border: none;
      
    }
    .content-tabs {
      flex-grow : 1;
    }
    .content {
      background-color:RGBA(43,45,66, 0.85);
      padding: 20px;
      width: 100%;
      height: 100%;
      display: none;
    }
    .content h2 {
      padding: 0px 0 5px 0px;
    }
    .content hr {
      width: 100px;
      height: 2px;
      background: #222;
      margin-bottom: 5px;
    }
    .content p {
      width: 100%;
      height: 100%;
    }
    .active-content {
      display: block;
    }
  } 
  
  @media all and (min-width: 425px){

    .homeWeather-card{
      width: 80vw;
    }

    .container {
      display: flex;
      flex-direction: column;
      position: relative;
      text-align: center;
      height: 25vh;
      width: 100vw;
      align-content: center;
      background-color:RGBA(43,45,66,0.65);
      color: #eae6ec;
      
   
    }
    
    .bloc-tabs {
      display: flex;
      width: 80vw;
    }
  
    .tabs {
      color: #eae6ec;
      padding: 2vw;
      text-align: center;
      width: 50%;
      background: rgba(128, 128, 128, 0.210);
      cursor: pointer;
      border-bottom: 1px solid rgba(0, 0, 0, 0.274);
      box-sizing: content-box;
      position: relative;
      outline: none;
    }
  
    .tabs:not(:last-child){
      border-right: 1px solid rgba(0, 0, 0, 0.274);
    }
    
    .active-tabs  {
      background-color:RGBA(43,45,66,0.85);;
      border-bottom: 1px solid transparent;
    }
    
    .active-tabs::before {
      content: "";
      display: block;
      position: absolute;
      top: -5px;
      left: 50%;
      transform: translateX(-50%);
      width: calc(100% + 2px);
      height: 5px;
      background: rgb(88, 147, 241);
    }
    
    

    button {
      border: none;
      
    }
    .content-tabs {
      flex-grow : 1;
    }
    .content {
      background-color:RGBA(43,45,66, 0.85);
      padding: 20px;
      width: 80vw;
      height: 100%;
      display: none;
    }
    .content h2 {
      padding: 0px 0 5px 0px;
    }
    .content hr {
      width: 100px;
      height: 2px;
      background: #222;
      margin-bottom: 5px;
    }
    .content p {
      width: 100%;
      height: 100%;
    }
    .active-content {
      display: block;
    }
  } 
  
  @media all and (min-width: 768px){

    .homeWeather-card{
      width: 80vw;
      height: 60vh;
    }

    .container {
      display: flex;
      flex-direction: column;
      position: relative;
      text-align: center;
      height: 25vh;
      width: 100vw;
      align-content: center;
      background-color:RGBA(43,45,66,0.65);
      color: #eae6ec;
      
      margin-bottom: 50vh;
    }
    
    .bloc-tabs {
      display: flex;
      width: 80vw;
    }
  
    .tabs {
      color: #eae6ec;
      padding: 2vw;
      text-align: center;
      width: 50%;
      background: rgba(128, 128, 128, 0.210);
      cursor: pointer;
      border-bottom: 1px solid rgba(0, 0, 0, 0.274);
      box-sizing: content-box;
      position: relative;
      outline: none;
    }
  
    .tabs:not(:last-child){
      border-right: 1px solid rgba(0, 0, 0, 0.274);
    }
    
    .active-tabs  {
      background-color:RGBA(43,45,66,0.85);;
      border-bottom: 1px solid transparent;
    }
    
    .active-tabs::before {
      content: "";
      display: block;
      position: absolute;
      top: -5px;
      left: 50%;
      transform: translateX(-50%);
      width: calc(100% + 2px);
      height: 5px;
      background: rgb(88, 147, 241);
    }
    
    

    button {
      border: none;
      
    }
    .content-tabs {
      flex-grow : 1;
    }
    .content {
      background-color:RGBA(43,45,66, 0.85);
      padding: 20px;
      width: 80vw;
      height: 100%;
      display: none;
    }
    .content h2 {
      padding: 0px 0 5px 0px;
    }
    .content hr {
      width: 100px;
      height: 2px;
      background: #222;
      margin-bottom: 5px;
    }
    .content p {
      width: 100%;
      height: 100%;
    }
    .active-content {
      display: block;
    }
  }
  
  
  @media all and (min-width: 1024px) {
    
    .homeWeather-card{
      width: 80vw;
      height: 60vh;
      
    }

    .container {
      display: flex;
      flex-direction: column;
      position: relative;
      text-align: center;
      height: 25vh;
      width: 100vw;
      align-content: center;
      background-color:RGBA(43,45,66,0.65);
      color: #eae6ec;
      
 
    }
    
    .bloc-tabs {
      display: flex;
      width: 80vw;
    }
  
    .tabs {
      color: #eae6ec;
      padding: 2vw;
      text-align: center;
      width: 50%;
      background: rgba(128, 128, 128, 0.210);
      cursor: pointer;
      border-bottom: 1px solid rgba(0, 0, 0, 0.274);
      box-sizing: content-box;
      position: relative;
      outline: none;
    }
  
    .tabs:not(:last-child){
      border-right: 1px solid rgba(0, 0, 0, 0.274);
    }
    
    .active-tabs  {
      background-color:RGBA(43,45,66,0.85);;
      border-bottom: 1px solid transparent;
    }
    
    .active-tabs::before {
      content: "";
      display: block;
      position: absolute;
      top: -5px;
      left: 50%;
      transform: translateX(-50%);
      width: calc(100% + 2px);
      height: 5px;
      background: rgb(88, 147, 241);
    }
    
    

    button {
      border: none;
      
    }
    .content-tabs {
      flex-grow : 1;
    }
    .content {
      background-color:RGBA(43,45,66, 0.85);
      padding: 20px;
      width: 80vw;
      height: 100%;
      display: none;
    }
    .content h2 {
      padding: 0px 0 5px 0px;
    }
    .content hr {
      width: 100px;
      height: 2px;
      background: #222;
      margin-bottom: 5px;
    }
    .content p {
      width: 100%;
      height: 100%;
    }
    .active-content {
      display: block;
    }
  }

  @media all and (min-width: 1440px){

    .homeWeather-card{
      
      width: 60vw;
      height: 75vh;
    
      
    }

    .container {
      
      display: flex;
      flex-direction: column;
      position: relative;
      text-align: center;
      height: 25vh;
      width: 100vw;
      align-content: center;
      background-color:RGBA(43,45,66,0.65);
      color: #eae6ec;
      
 
    }
    
    .bloc-tabs {
      display: flex;
      width: 60vw;
      height: 9vh;
    }
  
    .tabs {
      color: #eae6ec;
      padding: 2vw;
      text-align: center;
      width: 50%;
      background: rgba(128, 128, 128, 0.210);
      cursor: pointer;
      border-bottom: 1px solid rgba(0, 0, 0, 0.274);
      box-sizing: content-box;
      position: relative;
      outline: none;
      font-size: 0.9vw;
    }
  
    .tabs:not(:last-child){
      border-right: 1px solid rgba(0, 0, 0, 0.274);
    }
    
    .active-tabs  {
      background-color:RGBA(43,45,66,0.85);;
      border-bottom: 1px solid transparent;
    }
    
    .active-tabs::before {
      content: "";
      display: block;
      position: absolute;
      top: -5px;
      left: 50%;
      transform: translateX(-50%);
      width: calc(100% + 2px);
      height: 5px;
      background: rgb(88, 147, 241);
    }
    
    

    button {
      border: none;
      
    }
    .content-tabs {
      flex-grow : 1;
    }
    .content {
      background-color:RGBA(43,45,66, 0.85);
      width: 60vw;
      height: 100%;
      display: none;
    }
    .content h2 {
      padding: 0px 0 5px 0px;
    }
    .content hr {
      width: 100px;
      height: 2px;
      background: #222;
      margin-bottom: 5px;
    }
    .content p {
      width: 100%;
      height: 100%;
    }
    .active-content {
      display: block;
    }
  }

@media all and (min-width: 1920px){


  .homeWeather-card{
    width: 60vw;
    height: 50vh;

  }

  .container {
    display: flex;
    flex-direction: column;
    position: relative;
    text-align: center;
    height: 25vh;
    width: 100vw;
    align-content: center;
    background-color:RGBA(43,45,66,0.65);
    color: #eae6ec;
    

  }
  
  .bloc-tabs {
    display: flex;
    width: 60vw;
  }

  .tabs {
    color: #eae6ec;
    padding: 2vw;
    text-align: center;
    width: 60vw;
    background: rgba(128, 128, 128, 0.210);
    cursor: pointer;
    border-bottom: 1px solid rgba(0, 0, 0, 0.274);
    box-sizing: content-box;
    position: relative;
    outline: none;
    font-size: 1.2vw;
  }

  .tabs:not(:last-child){
    border-right: 1px solid rgba(0, 0, 0, 0.274);
  }
  
  .active-tabs  {
    background-color:RGBA(43,45,66,0.85);;
    border-bottom: 1px solid transparent;
  }
  
  .active-tabs::before {
    content: "";
    display: block;
    position: absolute;
    top: -5px;
    left: 50%;
    transform: translateX(-50%);
    width: calc(100% + 2px);
    height: 5px;
    background: rgb(88, 147, 241);
  }
  
  

  button {
    border: none;
    
  }
  .content-tabs {
    flex-grow : 1;
  }
  .content {
    background-color:RGBA(43,45,66, 0.85);
    padding: 20px;
    width: 60vw;
    height: 60vh;
    display: none;
  }
  .content h2 {
    padding: 0px 0 5px 0px;
  }
  .content hr {
    width: 100px;
    height: 2px;
    background: #222;
    margin-bottom: 5px;
  }
  .content p {
    width: 100%;
    height: 100%;
  }
  .active-content {
    display: block;
  }
}