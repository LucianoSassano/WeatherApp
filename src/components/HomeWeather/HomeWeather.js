import React, { useState, useEffect } from "react";
import Tab from "../../components/Tab/Tab.js"

const HomeWeather = (props) => {
  const [weather, setWeather] = useState([]);

  useEffect(() => {
    setWeather(props.forecast);
  }, [props.forecast]);

  const handleFavorite = () => {
    props.handleFav(weather);
  };

  return (
    <>
      {weather.main !== undefined ? (
        <div className=" container-data">
          <Tab/>
          <h1>
            {weather.name},{weather.sys.country}
          </h1>
          <h3>{Math.round(weather.main.temp)} Â°C</h3>
          <h3>{weather.weather[0].main}</h3>
          <h4>{weather.weather[0].description}</h4>
          <img
            src={`${process.env.REACT_APP_OPEN_WEATHER_IMAGE_URL}/img/wn/${weather.weather[0].icon}@2x.png`}
          ></img>
          <form className="button-form">
            <a
              data-testid="shareBtn"
              className="btn btn-primary"
              data-toggle="collapse"
              href="#collapseSocial"
              role="button"
              aria-expanded="false"
              aria-controls="collapseExample"
            >
              Share
            </a>
            <button
              data-testid="favoriteBtn"
              onClick={handleFavorite}
              type="button"
              className="btn btn-primary"
            >
              Fav
            </button>
          </form>
          
          <div className="collapse" id="collapseSocial">
            <iframe
              title="facebook share"
              className="facebook-btn"
              src={`${process.env.REACT_APP_FACEBOOK_SHARE_BTN_URL}?href=http%3A%2F%2Fapi.openweathermap.org%2Fdata%2F2.5%2Fweather%3Fq%3D${weather.name}%26units%3Dmetric%26appid%3De0587d27e5b47cb13ba1b093e60738f7&layout=button&size=large&width=103&height=28&appId`}
              width="103"
              height="28"
              scrolling="no"
              frameBorder="0"
              allowtransparency="true"
              allow="encrypted-media"
            ></iframe>
          </div>
        </div>
      ) : (
        ""
      )}
    </>
  );
};

export default HomeWeather;
