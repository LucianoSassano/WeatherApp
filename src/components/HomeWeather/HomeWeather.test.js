import React from 'react';
import { cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import HomeWeather from './HomeWeather.js';


afterEach(cleanup);


it('HomeWeather renders correctly', () => {

    const div = document.createElement("div");
    ReactDOM.render(<HomeWeather></HomeWeather>, div)
})




it("HomeWeather component matches snapshot", () => {

    const homeWeather = renderer.create(<HomeWeather></HomeWeather>).toJSON();
    expect(homeWeather).toMatchSnapshot();

})

