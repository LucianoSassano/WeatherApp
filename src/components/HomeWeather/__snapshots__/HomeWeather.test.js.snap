// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeWeather component matches snapshot 1`] = `""`;
